realsense2_camera/src/t265_realsense_node.cpp

```diff
@@ -6,7 +6,67 @@ T265RealsenseNode::T265RealsenseNode(ros::NodeHandle& nodeHandle,
                                      ros::NodeHandle& privateNodeHandle,
                                      rs2::device dev,
                                      const std::string& serial_no) : 
-                                     BaseRealSenseNode(nodeHandle, privateNodeHandle, dev, serial_no) {}
+                                     BaseRealSenseNode(nodeHandle, privateNodeHandle, dev, serial_no),
+                                     _wo_snr(dev.first<rs2::wheel_odometer>()),
+                                     _use_odom_in(false) 
+                                     {
+                                         initializeOdometryInput();
+                                     }
+
+void T265RealsenseNode::initializeOdometryInput()
+{
+    std::string calib_odom_file;
+    _pnh.param("calib_odom_file", calib_odom_file, std::string(""));
+    if (calib_odom_file.empty())
+    {
+        ROS_INFO("No calib_odom_file. No input odometry accepted.");
+        return;
+    }
+    std::ifstream calibrationFile(calib_odom_file);
+    if (not calibrationFile)
+    {
+        ROS_FATAL_STREAM("calibration_odometry file not found. calib_odom_file = " << calib_odom_file);
+        throw std::runtime_error("calibration_odometry file not found" );
+    }
+    const std::string json_str((std::istreambuf_iterator<char>(calibrationFile)),
+        std::istreambuf_iterator<char>());
+    const std::vector<uint8_t> wo_calib(json_str.begin(), json_str.end());
+
+    if (!_wo_snr.load_wheel_odometery_config(wo_calib))
+    {
+        ROS_FATAL_STREAM("Format error in calibration_odometry file: " << calib_odom_file);
+        throw std::runtime_error("Format error in calibration_odometry file" );
+    }
+    _use_odom_in = true;
+}
+
+void T265RealsenseNode::publishTopics()
+{
+    BaseRealSenseNode::publishTopics();
+    setupSubscribers();
+}
+
+void T265RealsenseNode::setupSubscribers()
+{
+    if (not _use_odom_in) return;
+
+    std::string topic_odom_in;
+    _pnh.param("topic_odom_in", topic_odom_in, DEFAULT_TOPIC_ODOM_IN);
+    ROS_INFO_STREAM("Subscribing to in_odom topic: " << topic_odom_in);
+
+    _odom_subscriber = _node_handle.subscribe(topic_odom_in, 1, &T265RealsenseNode::odom_in_callback, this);
+}
+
+void T265RealsenseNode::odom_in_callback(const nav_msgs::Odometry::ConstPtr& msg)
+{
+    ROS_INFO("Got in_odom message");
+    rs2_vector velocity {-(float)(msg->twist.twist.linear.y),

```

Comment:
No need to apply a coordinate transformation here. The relative transformation between odometry frame and T265 body frame is handled in the calibration.

Reply:
I want to keep the calibration.json file in a ROS standard. That is, X-forward, Y-Left and Z-Up.
For that reason it is necessary to convert from ROS coordinate system to T265's.